fendermint/app/src/cmd/genesis.rs

```diff
@@ -344,10 +352,116 @@ pub async fn seal_genesis(genesis_file: &PathBuf, args: &SealGenesisArgs) -> any
     builder.write_to(args.output_path.clone()).await
 }
 
+/// Fetches F3 certificate data from the parent Filecoin chain
+async fn fetch_f3_params_from_parent(
+    parent_endpoint: &url::Url,
+    parent_auth_token: Option<&String>,
+) -> anyhow::Result<Option<ipc::F3Params>> {
+    use std::convert::TryFrom;
+
+    tracing::info!(
+        "Fetching F3 certificate data from parent chain at {}",
+        parent_endpoint
+    );
+
+    let jsonrpc_client = JsonRpcClientImpl::new(
+        parent_endpoint.clone(),
+        parent_auth_token.map(|s| s.as_str()),
+    );
+
+    // We use a dummy subnet ID here since F3 data is at the chain level, not subnet-specific
+    let lotus_client = LotusJsonRPCClient::new(jsonrpc_client, SubnetID::default());
+
+    // Fetch F3 data using the Lotus client
+    match lotus_client.f3_get_instance_id().await {
+        Ok(instance_id) => {
+            tracing::info!("Found F3 instance ID: {}", instance_id);
+
+            // Get power table for this instance
+            let power_table_response = lotus_client.f3_get_power_table(instance_id).await?;
+
+            // Convert power entries
+            let power_table: anyhow::Result<Vec<_>> = power_table_response
+                .iter()
+                .map(|entry| {
+                    // Decode base64 public key
+                    let public_key_bytes = base64::Engine::decode(
+                        &base64::engine::general_purpose::STANDARD,
+                        &entry.pub_key,
+                    )?;
+                    // Parse the power string to u64
+                    let power = entry.power.parse::<u64>()?;
+                    Ok(types::PowerEntry {
+                        public_key: public_key_bytes,
+                        power,
+                    })
+                })
+                .collect();
+            let power_table = power_table?;
+
+            // Get latest certificate (optional)
+            let certificate = lotus_client.f3_get_certificate().await?;
```

Comment:
You already called `f3_get_certificate()` in `f3_get_instance_id()`.

Reply:
I didn't understand why not to fix this. Also, not sure it's even needed (see https://github.com/consensus-shipyard/ipc/pull/1438#discussion_r2472067081)